hyperspy/axes.py

```diff
@@ -343,8 +371,16 @@ def _get_index(self, value):
             return value
 
     def _parse_string_for_slice(self, value):
+        '''
+        Parses one of the start/stop/step parts of the slice
+        syntax. Returns integer if index, float if calibrated axis value.
+        '''
         if isinstance(value, str):
-            value = self._get_index_from_value_with_units(value)
+            # these are always returned as integers
+            if value[:3] == 'rel': # Slicing using relative syntax
```

Comment:
Matter of taste but I find `value.startswith('rel')` more readable.

Reply:
Totally agreed!